File: packages/skill-eval/src/runner/tool-defs.ts
import { z } from "zod";
import type { ToolDefinition } from "../providers/types.js";

export const defaultTools: ToolDefinition[] = [
	{
		name: "bash",
		description:
			"Run a bash command in the sandbox directory. Output is capped at 30KB. For large outputs, pipe through head, tail, or grep.",
		parameters: z.object({
			command: z.string(),
		}),
	},
	{
		name: "read_file",
		description:
			"Read the contents of a file. Files over 100KB are rejected — use bash with head/tail/grep to read specific parts of large files.",
		parameters: z.object({
			path: z.string(),
		}),
	},
	{
		name: "write_file",
		description: "Write content to a file (creates parent directories if needed)",
		parameters: z.object({
			path: z.string(),
			content: z.string(),
		}),
	},
	{
		name: "list_directory",
		description: "List files and directories at the given path",
		parameters: z.object({
			path: z.string(),
		}),
	},
];

File: packages/skillet-eval-python/src/skill_eval/runner/tool_defs.py
from __future__ import annotations

from pydantic import BaseModel

from skill_eval.providers.types import ToolDefinition


class BashParams(BaseModel):
    command: str


class ReadFileParams(BaseModel):
    path: str


class WriteFileParams(BaseModel):
    path: str
    content: str


class ListDirectoryParams(BaseModel):
    path: str


def _schema_from_model(model: type[BaseModel]) -> dict:
    schema = model.model_json_schema()
    return {
        "type": schema.get("type", "object"),
        "properties": schema.get("properties", {}),
        "required": schema.get("required", []),
    }


DEFAULT_TOOLS: list[ToolDefinition] = [
    ToolDefinition(
        name="bash",
        description="Run a bash command in the sandbox directory. Output is capped at 30KB. "
        "For large outputs, pipe through head, tail, or grep.",
        parameters=_schema_from_model(BashParams),
    ),
    ToolDefinition(
        name="read_file",
        description="Read the contents of a file. Files over 100KB are rejected — use bash "
        "with head/tail/grep to read specific parts of large files.",
        parameters=_schema_from_model(ReadFileParams),
    ),
    ToolDefinition(
        name="write_file",
        description="Write content to a file (creates parent directories if needed)",
        parameters=_schema_from_model(WriteFileParams),
    ),
    ToolDefinition(
        name="list_directory",
        description="List files and directories at the given path",
        parameters=_schema_from_model(ListDirectoryParams),
    ),
]
